### DISCUSSION TITLE
Google Cloud high (relatively) network costs for Nightscout

### DISCUSSION THREAD

**Discussion by BillShepp** [NightscoutFoundation/xDrip#3308] [Q&A] [answered]
I've had Nightscout running well on Google Cloud for about a month. Looking at my billing report I'm seeing daily costs of exactly $0.072, which appear to all be for network (compute is $0.00). After the three-month credit this will come to something like $2/month, which obviously isn't a lot but seems much higher than what the documentation suggests.

I can't imagine I'm generating an appreciable amount of traffic to NS - I have no followers, just my AAPS implementation, a copy of xDrip on my phone, and a NS menu bar widget on my laptop. I can't find a way in Google Cloud to get a report of the actual volume of network traffic, but the fact the cost is precisely the same every day suggests to me it's some sort of minimum or overhead charge rather than volume-based.

Anyone have any insight on this or suggestion on how I might reduce traffic to see if it reduces the network cost?

**Comment by Navid200** [comment, marked as answer]
Please look at this and after, if you have any questions, please let me know.
https://navid200.github.io/xDrip/docs/Nightscout/NS_FreeTier.html

As stated, the only way to logically talk about what you have been charged for is to create a cost table as explained.
Here again is how: https://navid200.github.io/xDrip/docs/Nightscout/CostTable.html
Please don't waste your time looking at the different charts in billing.

**Reply by BillShepp** [reply]
I missed this when reviewing the documentation. I don't yet have an invoice reflecting the daily network charges I'm seeing, but based on the December invoice (before I had AAPS running) I'm guessing those daily charges will be negated by the free tier. I'll post here again if that's not the case and things appear different than the documentation suggests.

**Comment by Navid200** [comment]
I know individuals who are charged up to $0.25 a month because of having many many many follower devices.
I can also imagine if you have followers that are really careless about minimizing unnecessary traffic, you may experience the same.
For example, if you  have a watch that keeps asking for backfill even though it has all the history!
Imagine if your watch asks for 3 hours worth of data once every 5 minutes.  That's a lot of redundant data being asked for for no good reason.

I have yet to see anyone showing a case like that, not caused by a mistake on their part in setup, where they are charged $0.50 or more a month.
